controllers/argocd/argocd_controller_test.go

```diff
@@ -102,6 +102,37 @@ func TestReconcileArgoCD_Reconcile(t *testing.T) {
 	}
 }
 
+func TestReconcileArgoCD_ReconcileLabel(t *testing.T) {
```

Comment:
Can you add few more test cases, where multiple argocd instances and label selector selects only few of them. Another test could be to test the default instance when label selector is not used, then all argocd instances will be reconciled.